meilisearch-auth/src/store.rs

```diff
@@ -23,19 +24,27 @@ pub type KeyId = [u8; KEY_ID_LENGTH];
 
 #[derive(Clone)]
 pub struct HeedAuthStore {
-    env: Env,
+    env: Arc<Env>,
     keys: Database<ByteSlice, SerdeJson<Key>>,
     action_keyid_index_expiration: Database<KeyIdActionCodec, SerdeJson<Option<DateTime<Utc>>>>,
 }
 
+impl Drop for HeedAuthStore {
+    fn drop(&mut self) {
+        if Arc::strong_count(&self.env) == 1 {
```

Comment:
I consider it safe to check that the count is equal to 1 as there is no way it can grow between the moment you acquire this number and you compare it to 1.